src/main/java/org/skriptlang/skript/bukkit/itemcomponents/equippable/elements/ExprEquipCompEntities.java

```diff
@@ -0,0 +1,120 @@
+package org.skriptlang.skript.bukkit.itemcomponents.equippable.elements;
+
+import ch.njol.skript.bukkitutil.EntityUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.components.EquippableComponent;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.itemcomponents.equippable.EquippableExperiment;
+import org.skriptlang.skript.bukkit.itemcomponents.equippable.EquippableWrapper;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@SuppressWarnings("rawtypes")
+@Name("Equippable Component - Allowed Entities")
+@Description("The entities allowed to wear the item. "
+	+ "NOTE: Equippable component elements are experimental. Thus, they are subject to change and may not work as intended.")
+@Example("set the allowed entities of {_item} to a zombie and a skeleton")
+@Example("""
+	set {_component} to the equippable component of {_item}
+	clear the allowed entities of {_component}
+	""")
+@RequiredPlugins("Minecraft 1.21.2+")
+@Since("INSERT VERSION")
+public class ExprEquipCompEntities extends PropertyExpression<EquippableWrapper, EntityData> implements EquippableExperiment {
+
+	static {
+		registerDefault(ExprEquipCompEntities.class, EntityData.class, "allowed entities", "equippablecomponents");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
+		setExpr((Expression<EquippableWrapper>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected EntityData @Nullable [] get(Event event, EquippableWrapper[] source) {
+		List<EntityData> types = new ArrayList<>();
+		for (EquippableWrapper wrapper : source) {
+			EquippableComponent component = wrapper.getComponent();
+			Collection<EntityType> allowed = component.getAllowedEntities();
+			if (allowed == null || allowed.isEmpty())
+				continue;
+			allowed.forEach(entityType -> {
+				Class<? extends Entity> entityClass = entityType.getEntityClass();
+				types.add(EntityData.fromClass(entityClass));
```

Comment:
I think you might prefer https://github.com/SkriptLang/Skript/blob/0b0b3da02e61ebb8019bb9d0922aaa8741731759/src/main/java/ch/njol/skript/bukkitutil/EntityUtils.java#L173